docs/minio-job.md

```diff
@@ -0,0 +1,209 @@
+                "Effect": "Allow",
+                "Action": [
+                    "s3:*"
+                ],
+                "Resource": [
+                    "arn:aws:s3:::memes",
+                    "arn:aws:s3:::memes/*"
+                ]
+            }
+        ]
+    }
+---
+apiVersion: job.min.io/v1alpha1
+kind: MinIOJob
+metadata:
+  name: minio-test-job
+spec:
+  mcImage: quay.io/minio/mc:latest # optional, defaults to minio/mc:latest
+  serviceAccountName: mc-job-sa
+  securityContext: {}
+  containerSecurityContext: {}
+  tenant:
+    name: mytest-minio
+  commands:
+    - op: make-bucket
+      args:
+        name: memes
+    - name: add-my-user-1
+      op: admin/user/add
+      args:
+        user: ${USER}
+        password: ${PASSWORD}
+      envFrom:
+        - secretRef:
+            name: mytestsecretenvs
+    - name: add-my-user-2
+      op: admin/user/add
+      args:
+        user: pedro
+        password: $PASSWORD
+      env:
+        - name: PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: mytestsecret
+              key: PASSWORD
+    - name: add-my-policy
+      op: admin/policy/create
+      args:
+        name: memes-access
+        policy: /temp/policy.json
+      volumeMounts:
+        - name: policy
+          mountPath: /temp
+      volumes:
+        - name: policy
+          configMap:
+            name: mytestconfig
+            items:
+              - key: policy.json
+                path: policy.json
+    - op: admin/policy/attach
+      dependsOn:
+        - add-my-user-1
+        - add-my-user-2
+        - add-my-policy
+      args:
+        policy: memes-access
+        user: daniel
+    - op: admin/policy/attach
+      dependsOn:
+        - add-my-user-1
+        - add-my-user-2
+        - add-my-policy
+      args:
+        policy: memes-access
+        user: pedro
+    - op: stat
+      command:
+        - "mc"
+        - "stat"
+        - "myminio/memes"
+```
+The MinioJob is a Kubernetes Job that runs mc commands. It uses the MinIO client (mc) to interact with the MinIO server.
+## mcImage
+The `mcImage` field specifies the Docker image that will be used to run the mc commands. In this case, the image is `minio/mc:latest`.
+## serviceAccountName
+The `serviceAccountName` field specifies the name of the Kubernetes ServiceAccount that will be used to run the mc commands. In this case, the ServiceAccount is `mc-job-sa`.
+## securityContext
+example:
+```yaml
+runAsUser: 1000
+runAsGroup: 1000
+fsGroup: 1000
+fsGroupChangePolicy: "OnRootMismatch"
+runAsNonRoot: true
+allowPrivilegeEscalation: false
+capabilities:
+  drop:
+    - ALL
+```
+The `securityContext` field specifies the security context that will be used to run the mc commands. 
+## containerSecurityContext
+The `containerSecurityContext` field specifies the security context that will be used to run the `mc` commands in the container.
+## tenant
+```yaml
+name: tenantName
+namespace: tenantNamespace
```

Comment:
I guess the `namespace` should only be set if the job is created in another namespace? If the tenant and job are in the same namespace, then it should not be necessary to specify it.

Reply:
Since this can allow siginficant changes to a cluster, I wonder if we should reqiure/validate that the job being run was applied in the namespace matching the target tenant

So if I did `kubectl apply minioJob.yaml -n foobar` there should be some sort of valiation error since we have namespace mismatch.

Reply:
@jiuker Do you always need to set the namespace or only if you want to address a tenant outside the namespace where the job is created?

Reply:
In Kubernetes, you mostly don't need to specify a namespace if you reference an object in your local namespace. I think we should reconsider.

Reply:
A valid point @ramondeklein maybe we could enforce "MinIO Jobs can only be created in the same tenant namespace", if this is somenthing we can add in a future release, then add it as a feature issue  to the MVP https://github.com/miniohq/engineering/issues/1302 @jiuker @ramondeklein

We don't expect v1alpha1 to be complete in first release, the purpose of this PR is to document current implementation, this can evolve with the project, It's a great win that this kind of conversations arised, thank you guys!

Reply:
@jiuker I see in the example that no `.tenant.namespace` was set, so I assume that it uses the tenant from the local namespace. But this contradicts with you statement [here](https://github.com/minio/operator/pull/2173/files/10b87da41f2a4cb087c2c34f12ecab2c35492c93#r1654890343).

Reply:
Okay, so `.tenant.namespace` is optional and defaults to the local namespace.